core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java

```diff
@@ -0,0 +1,1377 @@
+        long quotes = chunkStats.get(0);
+
+        indexChunkStats.setPos(0);
+        //set first chunk offset and line number
+        indexChunkStats.add(0);
+        indexChunkStats.add(0);
+
+        long lines;
+        long totalLines = chunks > 0 ? chunkStats.get(1) + 1 : 1;
+
+        for (int i = 1; i < chunks; i++) {
+            long startPos;
+            if ((quotes & 1) == 1) { // if number of quotes is odd then use odd starter
+                startPos = chunkStats.get(5 * i + 4);
+                lines = chunkStats.get(5 * i + 2);
+            } else {
+                startPos = chunkStats.get(5 * i + 3);
+                lines = chunkStats.get(5 * i + 1);
+            }
+
+            //if whole chunk  belongs to huge quoted string or contains one very long line
+            //then it should be ignored here and merged with previous chunk
+            if (startPos > -1) {
+                indexChunkStats.add(startPos);
+                indexChunkStats.add(totalLines);
+            }
+
+            quotes += chunkStats.get(5 * i);
+            totalLines += lines;
+        }
+
+        if (indexChunkStats.get(indexChunkStats.size() - 2) < fileLength) {
+            indexChunkStats.add(fileLength);
+            indexChunkStats.add(totalLines);//doesn't matter
+        }
+    }
+
+    private void processIndexStats() {
+        LongHashSet set = new LongHashSet();
+        for (int i = 0, n = partitionKeysAndSizes.size(); i < n; i += 2) {
+            set.add(partitionKeysAndSizes.get(i));
+        }
+
+        LongList distinctKeys = new LongList();
+        for (int i = 0, n = set.size(); i < n; i++) {
+            distinctKeys.add(set.get(i));
+        }
+        distinctKeys.sort();
+
+        LongList totalSizes = new LongList();
+        for (int i = 0, n = distinctKeys.size(); i < n; i++) {
+            long key = distinctKeys.getQuick(i);
+            long size = 0;
+
+            for (int j = 0, m = partitionKeysAndSizes.size(); j < m; j += 2) {
+                if (partitionKeysAndSizes.getQuick(j) == key) {
+                    size += partitionKeysAndSizes.get(j + 1);
+                }
+            }
+
+            totalSizes.add(size);
+        }
+
+        DateFormat dirFormat = PartitionBy.getPartitionDirFormatMethod(partitionBy);
+
+        for (int i = 0, n = distinctKeys.size(); i < n; i++) {
+            long key = distinctKeys.getQuick(i);
+            long size = totalSizes.getQuick(i);
+
+            partitionNameSink.clear();
+            dirFormat.format(distinctKeys.get(i), null, null, partitionNameSink);
+            String dirName = partitionNameSink.toString();
+
+            partitions.add(new PartitionInfo(key, dirName, size));
+        }
+    }
+
+    private void processInner() throws TextException {
+        if (this.queue.getCycle() <= 0) {
+            throw TextException.$("Unable to process, the processing queue is misconfigured");
```

Comment:
consider a hint on what to do. as this error on its own is not very actionable. what queue? how can I fix it?

Reply:
Addressed in latest commit by throwing error  with better message . getCycle() returns queue size actually .